src/odemis/driver/simulated.py

```diff
@@ -462,3 +466,507 @@ def __init__(self, name, role, axes, ranges=None, choices=None, **kwargs):
             axes_dict[a] = d
 
         GenericComponent.__init__(self, name, role, vas=None, axes=axes_dict, **kwargs)
+
+
+class ParabolicMirrorRayTracer:
+    """
+    Simulates ray tracing for a parabolic mirror system with a lens and camera.
+
+    This class provides methods to generate and trace rays from a hemispherical source,
+    compute their intersection with a paraboloid, reflect them, refract them through a lens,
+    and finally map their positions on a camera plane. Useful for simulating optical systems
+    involving parabolic mirrors and lenses (SPARC2).
+    """
+
+    def __init__(self,
+            mirror: model.HwComponent,
+            stage: model.HwComponent,
+            image: Optional[model.DataArray] = None,
+            l_aligned: float = 0.0,
+            s_aligned: float = 0.0,
+            z_aligned: float = 0.0,
+            dl_misaligned: float = 30e-6,
+            ds_misaligned: float = -40e-6,
+            dz_misaligned: float = 100e-6
+        ):
+        """
+        Initializes the RayTracerParabola with default parameters for the paraboloid,
+        lens, and camera setup. Also sets the misalignment for the mirror and stage.
+        """
+        # Parabolic mirror properties (assume standard High-NA mirror)
+        self.a = 0.1
+        self.xcut = 10.75
+        self.dfoc = 0.5
+        self.holesize = 0.6
```

Comment:
Let's take these 4 values as arguments. Preferably, in meters (not mm), and with names as close as we use in the rest of odemis:
```
       x_max: 13.25e-3,  # xcut? 
       hole_diam: 0.6e-3,  # hole size
       focus_dist: 0.5e-3,  # dfoc
       parabola_f: 2.5e-3,  # a, =1/(4 * parabola_f)?
```